### PR TITLE
[ci] upgrade R CI scripts to work on Ubuntu 20.04

### PR DESCRIPTION
**Pull request by jameslamb**
For the last two days, R CI jobs on Ubuntu in this project have been broken, with an error like this from `R CMD check`.

```text
* checking PDF version of manual ...  WARNING
LaTeX errors when creating PDF version.
This typically indicates Rd problems.
LaTeX errors found:
! LaTeX Error: File `letltxmacro.sty' not found.

Type X to quit or <RETURN> to proceed,
or enter new name. (Default extension: sty)

! Emergency stop.
<read *> 
         
l.132 \RequirePackage{letltxmacro}[2008/06/13]
                                              ^^M
!  ==> Fatal error occurred, no output PDF file produced!
```

For example: https://github.com/microsoft/LightGBM/pull/3997/checks?check_run_id=2143509388

We've seen similar errors be resolved in the past just by waiting a bit for package managers to catch up. However, I think that this error is actually a permanent problem that needs a permanent fix (https://github.com/microsoft/LightGBM/pull/3997#issuecomment-802433009).

I tried googling for that error and found a forum post on askubuntu.com that said this can be solved on Ubuntu 20.04 by installing an additional TeX package bundle called `texlive-latex-extra` (https://askubuntu.com/questions/1253671/missing-letltxmacro-sty-after-latex-texstudio-installation).

I think the root cause of this problem is that `ubuntu-latest` images from GitHub Actions are now Ubuntu 20.04! We have been getting warnings about that on GitHub Actions and Azure DevOps for months, and I guess this week the change finally happened: https://github.com/actions/virtual-environments/issues/1816#issuecomment-801714505.

I see this in the job linked above.

```text
Current runner version: '2.277.1'
Operating System
  Ubuntu
  20.04.2
  LTS
```

This PR proposes installing that additional TeX dependency in R Linux jobs. Hopefully it will fix the CI issues.

### CODE REVIEW

**Comment by jameslamb** [comment]
I noticed this error on the CRAN job:

> E: Version '2.69-11' for 'autoconf' was not found

Guess that needs to be updated too, will do that right now.

**Comment by jameslamb** [comment]
ok next failure, the compiled size of the library gets a lot bigger using the Ubuntu 20.04 image from GitHub Actions.

```
* checking installed package size ... NOTE
  installed size is 82.5Mb
  sub-directories of 1Mb or more:
    libs  81.9Mb
```

` r-package (ubuntu-latest, gcc, R 4, cran) (pull_request)`:  https://github.com/microsoft/LightGBM/pull/4084/checks?check_run_id=2145403864

This could be caused by many things, like different compiler versions. Either way, it's not a problematic size. There are larger packages on CRAN. For example, `{duckdb}` is 165 MB: https://cran.r-project.org/web/checks/check_results_duckdb.html and `{torch}` is 120 MB: https://cran.r-project.org/web/checks/check_results_torch.html.

I'll update the allowed size as part of this PR.

File: R-package/README.md [hunk 1]
```diff
@@ -291,20 +291,20 @@ This section briefly explains the key files for building a CRAN package. To upda
...
     ```
 
-    If you do not have easy access to an Ubuntu 18.04 environment, the `configure` script can be generated using Docker by running the code below from the root of this repo.
+    If you do not have easy access to an Ubuntu 20.04 environment, the `configure` script can be generated using Docker by running the code below from the root of this repo.
 
     ```shell
     docker run \
         -v $(pwd):/opt/LightGBM \
-        -t ubuntu:18.04 \
+        -t ubuntu:20.04 \
```
Comment (jameslamb) [inline R-package/README.md:307]: for reviewers...I did re-run this, and `configure` was not changed at all.

**Comment by jameslamb** [comment]
@StrikerRUS based on our time zones I'll likely be asleep when you first see this. Since this is blocking other PRs, if you notice small things you'd like to change or fix you can just push directly to this branch (it's a LightGBM branch, not my fork) and merge without waiting on me.

**Review by StrikerRUS — APPROVED** [review]
Ah, R infrastructure is so fragile!.. Thanks a lot for fixing all issues!

**Comment by StrikerRUS** [comment]
BTW, did you see CRAN is testing M1mac? 😱 

https://github.com/RGF-team/rgf/pull/338#issuecomment-799716918
https://r.789695.n4.nabble.com/Apple-M1-CRAN-checks-td4767828.html#a4767910

**Pull request merged**